server_routers.go

```diff
@@ -350,6 +374,74 @@ func (d libp2pRouter) FindPeers(ctx context.Context, pid peer.ID, limit int) (it
 	return iter.ToResultIter(iter.FromSlice([]*types.PeerRecord{rec})), nil
 }
 
+func (d libp2pRouter) GetClosestPeers(ctx context.Context, key cid.Cid) (iter.ResultIter[*types.PeerRecord], error) {
+	// Per the spec, if the peer ID is empty, we should use self.
+	if key == cid.Undef {
+		key = peer.ToCid(d.host.ID())
+	}
+
+	keyStr := string(key.Hash())
+	var peers []peer.ID
+	var err error
+
+	switch v := d.routing.(type) {
+	case *dual.DHT:
+		// Only use WAN DHT for public HTTP Routing API.
+		// LAN DHT contains private network peers that should not be exposed publicly.
+		peers, err = v.WAN.GetClosestPeers(ctx, keyStr)
```

Comment:
ℹ️ This (WAN-only) feels like a safer behavior, than falling back to LAN on error, which could lead to mixing unrelated DHTs.